.github/workflows/wit.yaml

```diff
@@ -0,0 +1,59 @@
+name: Publish WIT package
+
+on:
+  push:
+    tags:
+    - 'wit-v*'
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+    - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871
+    - name: Extract tag context
+      id: ctx
+      run: |
+          version=${GITHUB_REF_NAME#wit-v}
+          echo "version=${version}" >> "$GITHUB_OUTPUT"
+          echo "tarball=wit-${version}.tar.gz" >> "$GITHUB_OUTPUT"
+          echo "version is ${version}"
+    - name: Download wkg
+      shell: bash
+      run: |
+        curl --fail -L https://github.com/bytecodealliance/wasm-pkg-tools/releases/download/v0.7.4/wkg-x86_64-unknown-linux-gnu -o wkg
+        chmod +x wkg;
+        echo "$(realpath .)" >> "$GITHUB_PATH";
+    - name: Generate and set wkg config
+      shell: bash
+      env:
+        WKG_OCI_USERNAME: ${{ github.repository_owner }}
+        WKG_OCI_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        cat << EOF > wkg-config.toml
+        [namespace_registries]
+        wasmcloud = "wasmcloud.com"
+        wrpc = "bytecodealliance.org"
+        wasi = "wasi.dev"
+
+        [registry."wasmcloud.com".oci]
+        auth = { username = "${WKG_OCI_USERNAME}", password = "${WKG_OCI_PASSWORD}" }
+        EOF
+        echo "WKG_CONFIG_FILE=$(realpath wkg-config.toml)" >> $GITHUB_ENV
```

Comment:
Can these values not be passed in as environment variables in the `wkg` steps below?

I'd rather we have `wkg-config.toml` in the root of the repository and the authentication variables were passed in to the wkg commands as environment variables instead, GitHub should take care of obfuscating the secrets from logs as appropriate.

Reply:
They cannot. The reason is that the client doesn't know which client it is going to use for any given registry until it resolves it from config. OCI and warg require different configuration, so those vars are not exposed. This is global config for the tool anyway as we don't yet support overrides from a local `wkg.toml` yet

Reply:
Got it, that's unfortunate, but I appreciate the context.